_posts/2017/2017-11-26-choose-uuids-for-model-ids-in-rails.md

```diff
@@ -0,0 +1,83 @@
+---
+title: 'Choose UUIDs for model ids in Rails'
+description: 'A little more complexity, a little more security'
+layout: article
+category: ruby
+image:
+  base: '2017/choose-uuids-for-model-ids-in-rails'
+  alt: 'Rainbow UUIDs'
+---
+
+Ruby on Rails has had the [ability](https://github.com/rails/rails/pull/21762) to use UUIDs ([Universally Unique IDentifiers](https://en.wikipedia.org/wiki/Universally_unique_identifier)) for ActiveRecord models since version 5.0.
+
+These are a native column type in PostgreSQL, there's more details on native Postgres types in the [Rails Guides](http://guides.rubyonrails.org/active_record_postgresql.html).
+
+## Instead of…
+
+…using Rails default incrementing integer IDs.
+
+
+## Use…
+
+…PostgreSQL’s UUID support.
+
+
+### Enable extension
+
+`bin/rails g migration enable_extension_for_uuid`
+
+```ruby
+class EnableExtensionForUuid < ActiveRecord::Migration[5.1]
+  def change
+    enable_extension 'pgcrypto' unless extension_enabled?('pgcrypto')
+  end
+end
+```
+
+Change the default column type for primary keys
+
+Configure your migration generator to set id: :uuid for new tables:
```

Comment:
Is the formatting correct here? Should "Change the default column..." be a heading?